### PR TITLE
[FIX] [15.0] base_iso3166 breaking tests if code is False

### PR DESCRIPTION
**Pull request by ghost**
The module currently breaks tests in "delivery_multi_destination".

I added a test if code is False before getattr.

https://github.com/OCA/delivery-carrier/actions/runs/7362481660/job/20040960822?pr=751

Bildschirmfoto 2023-12-30 um 08 20 17

### CODE REVIEW

File: base_iso3166/models/res_country.py [hunk 1]
```diff
@@ -29,13 +33,15 @@ class ResCountry(models.Model):
     def _compute_codes(self):
         for country in self:
```
Comment (pedrobaeza) [inline base_iso3166/models/res_country.py]:
```suggestion
        for country in self.filtered("code"):
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Don't add logger. It's annoying. And reduce the diff simply filtering on the for.

File: base_iso3166/models/res_country.py (hunk 1 above)
Comment (ghost) [reply base_iso3166/models/res_country.py]: A) in the case of code=False, it's not calculating  "code_alpha3" and "code_numeric" which will produce errors
-> I would need to handle this case outside of the loop which would make the code more complex.
or do I miss something?

Comment (pedrobaeza) [reply base_iso3166/models/res_country.py]: Do a previous `self.code_alpha3 = False`. And please squash everything in one commit.

Comment (pedrobaeza) [reply base_iso3166/models/res_country.py]: You are still not implementing this and changing the existing code, so CI has broken

File: base_iso3166/models/res_country.py [hunk 2]
```diff
@@ -27,18 +27,18 @@ class ResCountry(models.Model):
...
+        for country in self.filtered("code"):
             c = False
-            for country_type in ["countries", "historic_countries"]:
-                try:
-                    c = getattr(pycountry, country_type).get(alpha_2=country.code)
-                except KeyError:
-                    c = getattr(pycountry, country_type).get(alpha2=country.code)
-                if c:
-                    break
+            if country.code:
```
Comment (pedrobaeza) [inline base_iso3166/models/res_country.py]: Remove this `if` and restore the previous code, and you still have several commits.

**Comment by ghost** [comment]
now it should be a clean commit

File: base_iso3166/models/res_country.py [hunk 3]
```diff
@@ -27,18 +27,15 @@ class ResCountry(models.Model):
...
-        for country in self:
+        self.code_alpha3 = False
+        self.code_numeric = False
+        for country in self.filtered("code"):
             c = False
             for country_type in ["countries", "historic_countries"]:
                 try:
                     c = getattr(pycountry, country_type).get(alpha_2=country.code)
                 except KeyError:
-                    c = getattr(pycountry, country_type).get(alpha2=country.code)
```
Comment (pedrobaeza) [inline base_iso3166/models/res_country.py]: Why are you removing this part? It's for compatibility with old versions.

Comment (ghost) [reply base_iso3166/models/res_country.py]: it's back

Comment (pedrobaeza) [reply base_iso3166/models/res_country.py]: Incorrect indentation and the break is missing.

Comment (ghost) [reply base_iso3166/models/res_country.py]: added the break, where is an incorrect indentation?

**Review by pedrobaeza — APPROVED** [review]
Finally it's correct!

**Pull request merged**